server/src/main/java/org/elasticsearch/index/get/GetResult.java

```diff
@@ -244,21 +262,7 @@ public XContentBuilder toXContentEmbedded(XContentBuilder builder, Params params
             builder.field(_PRIMARY_TERM, primaryTerm);
         }
 
-        List<DocumentField> metaFields = new ArrayList<>();
-        List<DocumentField> otherFields = new ArrayList<>();
-        if (fields != null && !fields.isEmpty()) {
-            for (DocumentField field : fields.values()) {
-                if (field.getValues().isEmpty()) {
-                    continue;
-                }
-                if (field.isMetadataField()) {
-                    metaFields.add(field);
-                } else {
-                    otherFields.add(field);
-                }
-            }
-        }
-        for (DocumentField field : metaFields) {
+        for (DocumentField field : metaFields.values()) {
```

Comment:
Using .values() can result in changes in ordering. However, it looks like we already had this problem. I opened https://github.com/elastic/elasticsearch/issues/42307 to followup.